pylossless/pipeline.py

```diff
@@ -592,24 +595,31 @@ def flag_epoch_gap(self, raw):
         annots = marks_flag_gap(raw, self.config['epoch_gap']['min_gap_ms'])
         raw.set_annotations(raw.annotations + annots)
 
-    def run_ica(self, raw):
-        ica_kwargs = self.config['ica']['ica_args']
+    def run_ica(self, raw, run):
+        ica_kwargs = self.config['ica']['ica_args'][run]
         if 'max_iter' not in ica_kwargs:
             ica_kwargs['max_iter'] = 'auto'
         if 'random_state' not in ica_kwargs:
             ica_kwargs['random_state'] = 97
 
         epochs = self.get_epochs(raw)
-        self.ica = ICA(**ica_kwargs)
-        self.ica.fit(epochs)
+        if run == 'run1':
+            self.ica1 = ICA(**ica_kwargs)
+            self.ica1.fit(epochs)
+
+        elif run == 'run2':
+            self.ica2 = ICA(**ica_kwargs)
+            self.ica2.fit(epochs)
+            self.ic_labels = label_components(epochs, self.ica2,
+                                              method="iclabel")
 
```

Comment:
Always a good practice to include an "else" statement. In that case, I'd say something like 
```python
else:
     raise ValueError("The `run` arguments can only take 'run1' or 'run2' as value.")
```

Reply:
thanks for the tip! will add this.

Reply:
Hey @christian-oreilly I _think_ we have a problem to be addressed in `run_ica`. If we want to omit bad time segments prior to fitting the ICA (which I think we do) then we need to pass a raw instance into the `fit` method: see the `reject_by_annotation` parameter description in the `ica.fit` method here: https://mne.tools/stable/generated/mne.preprocessing.ICA.html#mne.preprocessing.ICA.fit

Reply:
@scott-huberty Sorry for not responding to that earlier. If you look at how we coded that, get_epochs(...) returns epochs that already reject any segments with bad data. I think this is OK, unless you are seeing something I don't.